source/draw/gpu/GpuSurface.ooc

```diff
@@ -62,9 +62,16 @@ GpuSurface: abstract class extends Canvas {
 		translation := this _toLocal * FloatTransform3D createTranslation(box leftTop x, box leftTop y, this focalLength) * this _toLocal
 		translation * toReference * FloatTransform3D createScaling(box size x / 2.0f, box size y / 2.0f, 1.0f)
 	}
-	_createTextureTransform: static func (imageSize: IntVector2D, box: IntBox2D) -> FloatTransform3D {
+	_createTextureTransform: static func (imageSize: IntVector2D, box: IntBox2D, flipVector: FloatVector2D) -> FloatTransform3D {
 		scaling := FloatTransform3D createScaling(box size x as Float / imageSize x, box size y as Float / imageSize y, 1.0f)
 		translation := FloatTransform3D createTranslation(box leftTop x as Float / imageSize x, box leftTop y as Float / imageSize y, 0.0f)
+		if (flipVector x equals(-1.0f)) {
+			translation a = - translation a
+			translation m = 1.0f
+		} else if (flipVector y equals(-1.0f)) {
```

Comment:
Ok, so I think you need to remove the `else`, because if `x == -1.0f` then the `else if` is never checked.